modules/generator/processor/spanmetrics/spanmetrics.go

```diff
@@ -124,16 +123,17 @@ func (p *Processor) Shutdown(_ context.Context) {
 }
 
 func (p *Processor) aggregateMetrics(resourceSpans []*v1_trace.ResourceSpans) {
+	resourceLabels := make([]string, 0)
+	resourceValues := make([]string, 0)
 	for _, rs := range resourceSpans {
 		// already extract job name & instance id, so we only have to do it once per batch of spans
 		svcName, _ := processor_util.FindServiceName(rs.Resource.Attributes)
 		jobName := processor_util.GetJobValue(rs.Resource.Attributes)
 		instanceID, _ := processor_util.FindInstanceID(rs.Resource.Attributes)
-		resourceLabels := make([]string, 0) // TODO move outside the loop and reuse?
-		resourceValues := make([]string, 0) // TODO don't allocate unless needed?
-
+		resourceLabels = resourceLabels[:0]
+		resourceValues = resourceValues[:0]
 		if p.Cfg.EnableTargetInfo {
-			resourceLabels, resourceValues = processor_util.GetTargetInfoAttributesValues(rs.Resource.Attributes, p.Cfg.TargetInfoExcludedDimensions)
+			resourceLabels, resourceValues = processor_util.GetTargetInfoAttributesValues(rs.Resource.Attributes, p.Cfg.TargetInfoExcludedDimensions, intrinsicLabels)
```

Comment:
When TargetInfo is enabled, this will still realloc the labels and values on every loop iteration.  It should be possible to pass them in to `GetTargetInfoAttributesValues` to reuse them, which would be a nice improvement in that mode.

Reply:
not sure what you mean. target info can be different from one resource level to another and this loop is at the resource level